anomaly-detection/README.MD

```diff
@@ -0,0 +1,224 @@
+
+The event patterns are tracked per interacting party (here simplified per source IP address) and are validated by a state machine. The state machine's states define what possible events may occur next, and what new states these events will result in.
+ 
+The final aggregated results are grouped under network id which acts as a network domain abstraction hosting multiple server machines.
+
+The aggregated results are sinked to Elastic Search for visualizing from Kibana user interface.
+
+The following diagram depicts the state machine used in this example.
+
+```
+           +--<a>--> W --<b>--> Y --<e>---+
+           |                    ^         |     +-----<g>---> TERM
+   INITIAL-+                    |         |     |
+           |                    |         +--> (Z)---<f>-+
+           +--<c>--> X --<b>----+         |     ^        |
+                     |                    |     |        |
+                     +--------<d>---------+     +--------+
+```
+
+## Getting Started
+
+
+### Building Pravega & Flink connectors
+
+Follow below steps to build and publish artifacts from source to local Maven repository:
+
+```
+$git clone https://github.com/pravega/pravega.git
+$./gradlew clean install
+
+$git clone https://github.com/pravega/flink-connectors.git
+$./gradlew clean install
+
+```
+Alternatively, follow the instructions from [here](http://pravega.io/docs/getting-started/) to pull from release repository.
+
+### Building and Running Example
+
+- Clone the repository (git clone https://github.com/pravega/pravega-samples.git)
+- Build samples `./gradlew clean build`
+- Create local distribution `./gradlew clean installDist`
+
+<b>To run the application using script (Embedded Flink App)</b>
+
+Navigate to `$PRAVEGA_SAMPLES_HOME/anomaly-detection/build/install/pravega-flink-anomaly-detection/`
+
+Update `conf/app.json` (see "Application Configuration" section) as appropriate and run below commands to create stream and publish/consume records.
+
+```
+`bin/anomaly-detection --configDir <APP_CONFIG_DIR> --mode <RUN_MODE>`
+
+where, 
+
+- `APP_CONFIG_DIR` is the directory that contains the `app.json` configuration file
+- `RUN_MODE` can take any of the 3 options (1, 2 or 3),
+  - 1 = Create pravega stream and Scope using info from configuration file
+  - 2 = Publish events to Pravega, 
+  - 3 = Run Anomaly detection, 
+```
+
+Alternatively, you could also use `gradle run` to run from `$PRAVEGA_SAMPLES_HOME` directory
+
+```
+`./gradlew anomaly-detection:run -PargsList="--configDir <APP_CONFIG_DIR> --mode <RUN_MODE>"`
+``` 
+
+<b>To run the application using script (On top of existing Flink cluster)</b>
+
+Navigate to `$PRAVEGA_SAMPLES_HOME/anomaly-detection/build/install/pravega-flink-anomaly-detection/`
+
+Update `conf/app.json` (see "Application Configuration" section) as appropriate and run below commands to create stream and publish/consume records.
+
+```
```

Comment:
I think we should eliminate the `run` script in favor of the `flink run` command, because some of the below instructions use `flink ...` anyway.